java-util/src/main/java/io/druid/java/util/common/parsers/Parser.java

```diff
@@ -28,24 +29,41 @@
 public interface Parser<K, V>
 {
   /**
-   * Parse a String into a Map.
+   * Initialize this parser for centralized batch processing of files like IndexTask.
+   */
+  default void startFileFromBeginning()
```

Comment:
Do we really need both this and `reset()`? I would think that we can get rid of `reset()` and instead, call `startFileFromBeginning()` every time a new file starts. imo, reset isn't used for anything else so it's needless to keep it.